### ISSUE TITLE
itemsadder fastcraft support

### ISSUE THREAD

**Issue by AcesGamingUK** [BenWoodworth/FastCraft#187]
Fastcraft support for [Itemsadder](https://www.spigotmc.org/resources/%E2%9C%85must-have%E2%9C%85-itemsadder%E2%9C%A8-custom-items-huds-guis-textures-3dmodels-emojis-blocks-wings-hats.73355/)
itemsadder adds custom items/tools/entities to the game via NBT editing a base item and custom resources
currently, fast craft only sees the base item and thus can cause some unintended crafting behavior

**Comment by AcesGamingUK** [comment]
also support for custom recipe plugins
such as
[Itemsadder](https://www.spigotmc.org/resources/%E2%9C%85must-have%E2%9C%85-itemsadder%E2%9C%A8-custom-items-huds-guis-textures-3dmodels-emojis-blocks-wings-hats.73355/) , [CustomCrafting](https://www.spigotmc.org/resources/customcrafting-advanced-custom-recipe-plugin-1-14-1-16-x-free.55883/)

**Comment by BenWoodworth** [comment]
Sorry, I forgot to respond!

Could you provide configs/screenshots for recipes that aren't working right, so it's easy for me to reproduce?

And could you create separate issues for Item Creator Plus and Custom Crafting, so they can be kept track of separately?

I think I'll have some time to look into these tomorrow to see what's going on.

**Comment by Ekmot276** [comment]
my custom recipe plugin is working fine with fast-craft, plugins like itemsadder uses a "non vanilla" way to realize their custom recipes which will be very hard to implement

**Comment by BenWoodworth** [comment]
Yeah, FastCraft will create a crafting inventory and trigger `PrepareItemCraftEvent​` and then `CraftItemEvent`, so any recipes that use those *should* work fine (or at least get rejected if FastCraft gets the recipe wrong). Anything else I'd need to hook into an API to get/check the recipe some other way

**Comment by onicalls** [comment]
Hi, FastCraft v3.2.10 is not working with [ItemsAdder](https://www.spigotmc.org/resources/%E2%9C%85must-have%E2%9C%85-itemsadder%E2%9C%A8custom-items-huds-guis-mobs-3dmodels-emojis-blocks-wings-hats-liquids.73355/). IA has a full [API](https://github.com/LoneDev6/API-ItemsAdder#-packages) if you can check it.
